src/phoenix/server/api/mutations/dataset_mutations.py

```diff
@@ -300,6 +305,112 @@ async def delete_dataset(
             )
         )
 
+    @strawberry.mutation
+    async def patch_dataset_examples(
+        self,
+        info: Info[Context, None],
+        input: PatchDatasetExamplesInput,
+    ) -> DatasetMutationPayload:
+        if not (patches := input.patches):
+            raise ValueError("Must provide examples to patch.")
+        example_ids = [
+            from_global_id_with_expected_type(patch.example_id, DatasetExample.__name__)
+            for patch in patches
+        ]
```

Comment:
Can we just sort this to begin with, instead of using a complicated case statement for sql?
```suggestion
        example_ids = sorted(
            from_global_id_with_expected_type(patch.example_id, DatasetExample.__name__)
            for patch in patches
        )
```

Reply:
```python
        example_ids, patches = map(
            list,
            zip(
                *sorted(
                    (
                        from_global_id_with_expected_type(
                            patch.example_id, DatasetExample.__name__
                        ),
                        patch,
                    )
                    for patch in patches
                )
            ),
        )
````

Reply:
Addressed in 50fb7d69c9a42afc05f1e77af30e02da5f134f9c